delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,816 @@
+# Copyright 2021 The SODA Authors.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WarrayANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import hashlib
+import time
+
+import six
+
+from oslo_log import log as logging
+
+from delfin import exception, utils
+from delfin.common import constants
+from delfin.drivers.utils.ssh_client import SSHPool
+from delfin.drivers.hitachi.hnas import constants as constant
+from delfin.drivers.utils.tools import Tools
+
+LOG = logging.getLogger(__name__)
+
+
+class NasHandler(object):
+
+    def __init__(self, **kwargs):
+        self.ssh_pool = SSHPool(**kwargs)
+        self.evs_list = []
+
+    @staticmethod
+    def get_size(limit, is_calculate=False):
+        if limit == '0B':
+            return 0
+        if limit == '-':
+            return 0 if is_calculate else '-'
+        return int(Tools.get_capacity_size(limit))
+
+    def ssh_do_exec(self, command_list):
+        res = ''
+        res = self.ssh_pool.do_exec_command(command_list)
+        while 'Failed to establish SSC connection' in res:
```

Comment:
What would happend if device always return 'Failed to establish SSC connection'

Reply:
The SSH command line of the HNAS must be first accessed to the Linux server of the HNAS and then to the SSC subsystem of Linux. If the SSC subsystem cannot be accessed due to network fluctuation or other reasons, the SSH command fails and the system displays 'Failed to establish SSC connection'. In this case, run the command again to re-establish the connection with the SSC subsystem.

Reply:
The SSH command line of the HNAS must be first accessed to the Linux server of the HNAS and then to the SSC subsystem of Linux. If the SSC subsystem cannot be accessed due to network fluctuation or other reasons, the SSH command fails and the system displays 'Failed to establish SSC connection'. In this case, run the command again to re-establish the connection with the SSC subsystem.

Reply:
I mean if ssh always return this, it would be a dead loop

Reply:
We are considering setting a retry count or timeout